src/common_subset.rs

```diff
@@ -138,7 +188,9 @@ impl<NodeUid: Clone + Debug + Display + Eq + Hash + Ord> CommonSubset<NodeUid> {
         input_result.map(|mut queue| {
             if let Some(agreement_message) = opt_message {
                 // Append the message to agreement nodes to the common output queue.
-                queue.push_back(Output::Agreement(agreement_message))
+                queue.push_back(
+                    Target::All.message(Message::Agreement(proposer_id.clone(), agreement_message)),
```

Comment:
Here we have the proposer ID back as part of the message. Is this better than adding the proposer ID at the top level? The proposer ID is associated to the instance at the top level anyway.